File: ReSpect_v2/LAOS/Various.r
# Function to calculate tau and g values
taugcal <- function(p, Gp1, r, taumin, N) {
  # Initialize vectors
  taui <- numeric(N)
  H <- numeric(N)
  gi <- numeric(N)
  
  # Calculate tau values
  taui[1] <- taumin
  for (i in 2:N) {
    taui[i] <- taui[i-1] * r
  }
  
  # Calculate G1 and H values
  G1 <- 2 * Gp1 * gamma(1 + p) / p / pi * sin(p * pi / 2)
  H <- G1 * taui^(-p) / gamma(p)
  
  # Calculate g values
  gi <- H * (r^(p/2) - r^(-p/2)) / p
  
  return(list(taui = taui, gi = gi))
}

# Function to calculate Eabz
Eabzcal <- function(a, b, z, k) {
  return(z^k / gamma(a * k + b))
}

# Function to calculate R2
CalcR2 <- function() {
  # Set file paths
  FilePath <- file.path(getwd(), "WWF_62%_LAOS")
  FilePath2 <- getwd()
  
  # Define omega and amplitude vectors
  omegaV <- c(1, 10, 50, 100)
  AmpV <- c(0.05, 0.25, 0.5, 1.0, 2.0)
  
  # Initialize error variables
  error <- 0.0
  ErrorAroundMean <- 0.0
  R2 <- 0.0
  
  # Loop through amplitudes and frequencies
  for (j in rev(seq_along(AmpV))) {
    for (i in seq_along(omegaV)) {
      omega <- omegaV[i]
      Amp <- AmpV[j]
      AmpP <- Amp * 100
      
      # Construct file names
      FileName <- sprintf("Exp_WGF62%%_%d%%Om%d.txt", AmpP, omega)
      FileName2 <- sprintf("Theo_WGF62%%_%d%%Om%d.txt", AmpP, omega)
      
      # Load experimental and theoretical data
      q <- read.table(file.path(FilePath, FileName))
      qtheo <- read.table(file.path(FilePath2, FileName2))
      
      # Interpolate theoretical values
      xinterp <- seq_len(nrow(q))
      theoElStrvals <- approx(0:(nrow(qtheo)-1), qtheo[,5] / max(qtheo[,2]), 
                              xout = (xinterp-1) * (nrow(qtheo)-1) / nrow(q))$y
      
      # Calculate errors for elastic stress
      error <- error + sum((theoElStrvals - q[,5] / max(q[,2]))^2) / nrow(q)
      MeanEl <- mean(q[,5] / max(q[,2]))
      ErrorAroundMean <- ErrorAroundMean + sum((MeanEl - q[,5] / max(q[,2]))^2) / nrow(q)
      R2 <- R2 + 0.5 * (1 - error / ErrorAroundMean)
      
      # Interpolate theoretical values for viscous stress
      theoVisStrvals <- approx(0:(nrow(qtheo)-1), qtheo[,4] / max(qtheo[,2]), 
                               xout = (xinterp-1) * (nrow(qtheo)-1) / nrow(q))$y
      
      # Calculate errors for viscous stress
      error <- error + sum((theoVisStrvals - q[,4] / max(q[,2]))^2) / nrow(q)
      MeanVis <- mean(q[,4] / max(q[,2]))
      ErrorAroundMean <- ErrorAroundMean + sum((MeanVis - q[,4] / max(q[,2]))^2) / nrow(q)
      R2 <- R2 + 0.5 * (1 - error / ErrorAroundMean)
    }
  }
  
  # Calculate final error and R2 values
  error <- error / length(AmpV) / length(omegaV) / 2
  R2 <- R2 / length(AmpV) / length(omegaV)
  
  return(R2)
}

File: ReSpect_v2/LAOS/WWF_30_8_FMM/Various.m
function [taui, gi] =taugcal(p, Gp1, r, taumin, N)
taui = zeros(1, N);
H = zeros(1, N);
gi = zeros(1,N);
taui(1) = taumin;
for i=1:N-1
    taui(i+1) = taui(i)*r;
end
G1 = 2*Gp1*gamma(1+p)/p/pi*sin(p*pi/2);
for i = 1:N
    H(i) = G1*taui(i)^(-p)/gamma(p);
    gi(i) = H(i)*(r^(p/2) - r^(-p/2))/p;
end
end
    

function [Eabz] =Eabzcal(a, b, z, k)

Eabz = z.^k ./gamma(a .*k + b);

end
    

function R2 = CalcR2()



FilePath=[strcat(pwd,'\WWF_62%_LAOS\')];

FilePath2 = [strcat(pwd, '\')];




omegaV = [1, 10,50, 100];
AmpV = [0.05, 0.25, 0.5, 1.0, 2.0];


error = 0.0; 
ErrorAroundMean = 0.0;
R2 = 0.0;






for j=size(AmpV, 2):-1:1
for i=1:size(omegaV, 2)
    
        

        
        
omega = omegaV(i); Amp = AmpV(j);
AmpP = Amp*100;

FileName = strcat('Exp_WGF62%_', num2str(AmpP), '%Om',num2str(omega), ...
    '.txt' );

FileName2 = strcat('Theo_WGF62%_', num2str(AmpP), '%Om',num2str(omega), ...
    '.txt' );

q = load(strcat(FilePath, FileName) );
qtheo = load(strcat(FilePath2, FileName2) );



xinterp = [1:1:size(q, 1)];
theoElStrvals = (interp1([0:1:size(qtheo)-1], qtheo(:, 5) ./ ...
    max(qtheo(:, 2) ), (xinterp-1).*(size(qtheo,1)-1)./size(q,1) ) )';
error = error+ sum((theoElStrvals-q(:, 5)./ max(q(:,2) )).^2 )/size(q,1);

MeanEl = mean(q(:, 5)./ max(q(:,2) ) );
ErrorAroundMean = ErrorAroundMean + sum((MeanEl ...
    -q(:, 5)./ max(q(:,2) )).^2 )/size(q,1);

R2 = R2 + 0.5*(1-error/ErrorAroundMean);


theoVisStrvals = (interp1([0:1:size(qtheo)-1], qtheo(:, 4) ./ ...
    max(qtheo(:, 2) ), (xinterp-1).*(size(qtheo,1)-1)./size(q,1) ) )';

error = error+ sum((theoVisStrvals-q(:, 4)./ max(q(:,2) ) ).^2  )/size(q,1);

MeanVis = mean(q(:, 4)./ max(q(:,2) ));

ErrorAroundMean = ErrorAroundMean + sum((MeanVis ...
    -q(:, 4)./ max(q(:,2) ) ).^2  )/size(q,1);

R2 = R2 + 0.5*(1-error/ErrorAroundMean);


    end
end


% create plots elastic



error = error/size(AmpV, 2)/size(omegaV, 2)/2;

R2 = R2/size(AmpV, 2)/size(omegaV, 2);

% disp(['error = ', num2str(sqrt(error) )] );


end
